src/content/index.js

```diff
@@ -1,72 +1,141 @@
-
 'use strict';
 
 import { onError } from '../utils';
 import './index.css';
 
-const autoScrolling = {
-  tid: -1,
+const getScrollingElement = () => {
+  return document.scrollingElement
+    ? document.scrollingElement
+    : document.documentElement;
+};
+
+const defaultValues = {
+  intervalId: -1,
   x: 0,
   y: 0,
+  scrollingElement: null,
+  currentlyHovering: false,
   scrollingStep: 1,
   scrollingSpeed: 50,
-  scrollingElement: document.documentElement,
   stopScrollingByClick: true,
-  start: function () {
+  stopScrollingOnHover: false
+};
+
+let autoScrolling = Object.assign({}, defaultValues, {
+  scrollingAction: function() {
+    if (this.stopScrollingOnHover && this.currentlyHovering) {
+      return;
+    }
     this.y = this.y + this.scrollingStep;
     this.scrollingElement.scroll(this.x, this.y);
-    this.tid = setTimeout(() => {
```

Comment:
@jpsouzasilva 
> Seems like #18 is still happening even after changing to setInterval. : - | I'll be looking to fix that in a separate branch after this.
Okay. I want you to remove the changes about setTimeout/setInterval.
I want a PR that contains as only commits related this issue as possible.

Reply:
setInterval is just a more suitable API for ongoing recurring events. Using setTimeout continuously could lead to synchronization problems where two callbacks would trigger at the same time or be delayed for some reason, where as the interval is managed by the browser's update cycle itself.

Another issue with setTimeout would be if the timeout is delayed enough that the scrolling action would overwrite the last timeout before executing it.

The change to interval isn't for solving that issue, it just makes more sense in my opinion to have this.